### ISSUE TITLE
ToolTip issues prior to .NET 6.0

### ISSUE THREAD

**Issue by SamBent** [dotnet/wpf#4612]
This issue contains references to a number of ToolTip issues.  It is intended as a convenience for search, and a collection point for common information.  Please limit discussion to topics common to the collection as a whole;  comments about the individual bugs should go into the appropriate linked issue.

The issues collected here are
- Issues present in .NET 4.8 and above (i.e. .NET Core 3.x, .NET 5.x, previews of .NET 6.0)
- Actual bugs, as opposed to design changes
- Contrary to, or blocking of, the new ToolTip behavior mandated by the WCAG 2.1 compliance work.  See the [spec](https://github.com/dotnet/wpf/blob/main/src/Microsoft.DotNet.Wpf/specs/tooltip.md)
- Fixed by that work (in .NET 6.0)
- Not planned to be fixed in any earlier versions

Here is a sample app that illustrates most of the issues (see individual bugs for repro steps):
[WpfApp3.zip](https://github.com/dotnet/wpf/files/6650174/WpfApp3.zip)
The app artificially sets all tooltips to delay for 5 seconds before opening, and to close after displaying for 15 seconds.  This makes it easier to see the behavior, but does require a little patience.

**Comment by SamBent** [comment]
Here are the issues (not all of which appear in the mentions you probably see above):

- **ToolTip shows when owner acquires focus via non-keyboard #4613**.   Moving between apps, minimizing/restoring an app, and other non-keyboard ways of changing focus, should not show the tooltip.
- **Moving focus to an element with no ToolTip opens an ancestor's ToolTip #4614**.   Moving focus should not open a different element's tooltip.
- **Ctrl+Shift+F10 waits before opening the ToolTip #4615**.  Ctrl+Shift+F10 should toggle the tooltip immediately.
- **Mouse motion closes a ToolTip shown by keyboard #4616**.   Unrelated mouse motion should not close the tooltip.
- **Moving keyboard focus while a ToolTip is open shows the new ToolTip only after a delay #4617**.  Keyboard should work the same as mouse - a new tooltip shows immediately if another tooltip was already showing.
- **ToolTip toggles on Ctrl+Shift+Alt+F10 #4618**.  Extra modifier keys shouldn't toggle the tooltip.
- **BetweenShowDelay=0 does not work #4619**.  Setting BetweenShowDelay to 0 should force the next tooltip to delay before opening.
- **Tooltips show on keyboard focus #4172**.  In 4.8 et al., controls built-in to WPF could opt out of the "keyboard focus shows tooltip" feature, but apps and 3rd-party controls could not.  There should be a public, extensible, flexible way to do this.

**Comment by SamBent** [comment]
All these issues are fixed by the tooltip work in 6.0-preview7.

**Comment by taquion87** [comment]
Is [this](https://github.com/dotnet/wpf/issues/5730) something worth investigating?

**Comment by taquion87** [comment]
#5730

**Comment by taquion87** [comment]
I think #5730 should be included in this issues collection

**Comment by amirburbea** [comment]
I made a repro of this bug: https://github.com/amirburbea/PopupRepro

Run this app under .NET 5 and .NET 6 and you'll see that ListBoxItem has no issues with tooltip, but ComboBoxItem does.
